### DISCUSSION TITLE
How can we do batch update of useState

### DISCUSSION THREAD

**Discussion by numandev1** [react-native-community/discussions-and-proposals#627] [Ideas]
reference: https://github.com/reactwg/react-18/discussions/21

# What is batching?
Batching is when React **groups multiple state updates into a single re-render** for better performance.

how can we achieve batch updates in React Native?

**Comment by AwaisIbrar8** [comment]
I think the framework handles batches itself. 
React uses a concept called "reconciliation" to optimize updates to the UI. When you update the state of a component, React will batch multiple updates together and perform them at once, rather than updating the UI immediately after each change.

**Reply by numandev1** [reply]
but in react native batching is not occuring

**Reply by AwaisIbrar8** [reply]
I didn't check it. But I am thinking that it's happening. May be wrong mentioned in the documentation

**Reply by numandev1** [reply]
i checked it, batching is working on react web but not working in react native side. may be we have to use fabric renderer👀

**Comment by Grohden** [comment]
Seems react native doesn't have support for it yet, I still see `unstable_batchedUpdates` available in `0.73.4`, even though we're at react `18.2.0`

Its pretty simple to test if batching is working:
```tsx
import React, { useEffect, useState } from 'react';
import { Text, View, unstable_batchedUpdates } from 'react-native';

export const Test = () => {
  const [counterOne, setCounterOne] = useState(0);
  const [counterTwo, setCounterTwo] = useState(0);

  useEffect(() => {
    const interval = setInterval(() => {
      console.log('---------------');
      // To batch just:
      // unstable_batchedUpdates(() => {
      setCounterOne((x) => x + 1);
      setCounterTwo((x) => x + 1);
      // });
    }, 1000);

    return () => {
      clearInterval(interval);
    };
  }, []);

  useEffect(() => {
    console.log('Counter: ', counterOne, counterTwo);
  }, [counterOne, counterTwo]);

  return (
    <View
      style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
      }}>
      <Text>Counter One: {counterOne}</Text>
      <Text>Counter Two: {counterTwo}</Text>
    </View>
  );
};
```

Also, I'm not sure if having hermes affects that.. but we're with hermes disabled

**Comment by vksgautam1986** [comment]
New Arch supports it
